File: rust_study/rust_luogu/rust_codeforces/cpp/luogu/old/P2676.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P2676.cpp
 *@data      2024/01/17 21:24:45
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

bool cmp(int a, int b) {
    return a > b;
}

void solve() {
    int n, ans, t = 0;
    cin >> n >> ans;
    vector<int>qwq(n);
    for(auto &x : qwq)cin >> x;
    sort(qwq.begin(), qwq.end(), cmp);
    while(ans > 0)ans -= qwq[t++];
    cout << t;
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/old/p2676.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p2676.rs
//@data      2024/01/18 12:55:32
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let (a, b) = cin()
        .split_whitespace()
        .fold((0, 0), |x, y| (x.1, y.parse::<usize>().unwrap()));
    let mut qwq = Vec::new();
    (0..a).for_each(|_| qwq.push(cin().parse::<usize>().unwrap()));
    qwq.sort_unstable_by(|a, b| b.cmp(a));
    let mut ans = 0;
    for (i, &value) in qwq.iter().enumerate() {
        ans += value;
        if ans >= b {
            println!("{}", i + 1);
            break;
        }
    }
}
